Observer/RestApiUserObserver.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+
+class RestApiUserObserver implements ObserverInterface
+{
+    /**
+     * ManagerInterface
+     * 
+     * @var messageManager
+     */
+    protected $messageManager;
+
+    /**
+     * DataHelper
+     * 
+     * @var data_helper
+     */
+    protected $data_helper;
+
+    /**
+     * UserFactory
+     * 
+     * @var _userFactory
+     */
+    protected $_userFactory;
+
+    /**
+     * RoleCollectionFactory
+     * 
+     * @var RoleCollectionFactory
+     */
+    protected $_roleCollectionFactory;
+
+    /**
+     * Init
+     *
+     * @param MessageManager $messageManager
+     * @param DataHelper $data_helper
+     * @param UserFactory $_userFactory
+     * @param RoleCollectionFactory $_roleCollectionFactory
+     */
+    public function __construct(
+        \Magento\Framework\Message\ManagerInterface $messageManager,
+        \Klaviyo\Reclaim\Helper\Data $data_helper,
+        \Magento\User\Model\UserFactory $userFactory,
+        \Magento\Authorization\Model\ResourceModel\Role\CollectionFactory $roleCollectionFactory
+    ) {
+        $this->messageManager = $messageManager;
+        $this->data_helper = $data_helper;
+        $this->_userFactory = $userFactory;
+        $this->_roleCollectionFactory = $roleCollectionFactory;
+    }
+
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $apiUsername = $this->data_helper->getRestApiUsername();
+        $apiPassword = $this->data_helper->getRestApiPassword();
+        $apiEmail = $this->data_helper->getRestApiEmail();
+
+        //
+        $role = $this->_roleCollectionFactory->create();
+        $roleCollection = $role->getData();
+
+        $adminInfo = [
+            'role_id' => 1,
+            'username' => $apiUsername,
+            'firstname' => 'klaviyo',
+            'lastname'    => 'klaviyo',
+            'email'     => $apiEmail,
+            'password'  => $apiPassword,
+            'interface_locale' => 'en_US',
+            'is_active' => 1
+        ];
+
+        //try to get the ID of the Klaviyo role
+        try {
+            foreach ($roleCollection as $item) {
+                if ($item['role_name'] == 'Klaviyo' && $item['role_type'] == 'G') {
```

Comment:
constants for G and klaviyo, is there a reason it's capital K but from above, if so could we put another const (all string literals should be consts)

Reply:
ahh i see this is in the install script